src/mempool.rs

```diff
@@ -127,8 +154,30 @@ impl Mempool {
         match BlstCrypto::verify(&msg) {
             Ok(valid) => {
                 if valid {
```

Comment:
as you are adding more logic to this function, I would recommend to try having less indented. I like the "[guard clauses](https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a)" 

you could do this specific part as in consensus.rs: 
```rust
        if !BlstCrypto::verify(msg)? {
            self.metrics.signature_error("mempool");
            bail!("Invalid signature for message {:?}", msg);
        }
```

this is clearly optional :)

Reply:
I kept the match but used Ok(true) and Ok(false) to separate the cases.